### ISSUE TITLE
Switch to containerd (or CRI-O) in Galaxy

### ISSUE THREAD

**Issue by rkevin-arch** [LibreTexts/metalc#208]
I've talked about it in https://github.com/LibreTexts/protogalaxy/issues/2, but for the sake of tracking all issues in metalc, I'm also mentioning it here. Let me know if I should just move that issue over here.

**Comment by moorepants** [comment]
For now, let's keep all issues here (except for ckeditor thebe plugin). We can decide to distribute issues among repos if we want, but let's discuss that at a meeting.

**Comment by rkevin-arch** [comment]
OK. I opened it in protogalaxy since that's where we would deploy the fix (in `packages.pp`), so we can discuss there for now unless you think here's a better place.

**Comment by rkevin-arch** [comment]
Update: We have successfully used containerd to spawn a cluster. Everything works except Binder. I initially thought Binder can work with DinD, but DinD seems to want to mount the host Docker socket for some reason. This means we cannot use Binder without Docker on the host for now. I've [opened an issue in BinderHub](https://github.com/jupyterhub/binderhub/issues/1228) about this, and if no one knows how to deal with it I can look for a solution (either making repo2docker work without docker (maybe [something like this](https://github.com/vmware-tanzu/buildkit-cli-for-kubectl)) or running DinD inside containerd).

**Comment by rkevin-arch** [comment]
Update: Got everything to work. See https://github.com/jupyterhub/binderhub/issues/1228#issuecomment-744008333. Gonna run through the setup with everyone in our next meeting, and I think we can close this issue (unless we also want to try CRI-O).

**Comment by celine168** [comment]
Did some more research. I don't think the difference between containerd vs. CRI-O is too significant for us.

https://www.inovex.de/blog/containers-docker-containerd-nabla-kata-firecracker/

To paraphase...

It looks like containerd uses a shim, cri-containerd to make sure that it is CRI compliant. I believe it also fulfills the OCI specification as well (at least runc).

CRI-O was built to be lightweight and to implement CRI directly. It can work with any OCI runtime compliant software. I suppose it was built specifically for Kubernetes. 

tldr: CRI-O seems to be built specifically for Kubernetes and from the article I read and focuses on implementing the standards for OCI and CRI. I don't see a huge advantage over one or the other.

**Comment by pmackle** [comment]
This was fixed
